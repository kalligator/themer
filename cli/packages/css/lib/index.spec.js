const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('themer css theme generator', () => {
  const promisedFiles = Promise.all(render(colors));

  it('should generate valid CSS', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(3);
    files.forEach((file) => {
      const contents = file.contents.toString('utf8');
      expect(contents).toMatchSnapshot();
    });
  });

  it('should provide usage instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
