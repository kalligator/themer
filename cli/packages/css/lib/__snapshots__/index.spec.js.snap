// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer css theme generator should generate valid CSS 1`] = `
"
:root {
  --dark-accent0: #FF4050;

  --dark-accent1: #F28144;

  --dark-accent2: #FFD24A;

  --dark-accent3: #A4CC35;

  --dark-accent4: #26C99E;

  --dark-accent5: #66BFFF;

  --dark-accent6: #CC78FA;

  --dark-accent7: #F553BF;

  --dark-shade0: #282629;

  --dark-shade1: #474247;

  --dark-shade2: #656066;

  --dark-shade3: #847E85;

  --dark-shade4: #A29DA3;

  --dark-shade5: #C1BCC2;

  --dark-shade6: #E0DCE0;

  --dark-shade7: #FFFCFF;

  --light-accent0: #F03E4D;

  --light-accent1: #F37735;

  --light-accent2: #EEBA21;

  --light-accent3: #97BD2D;

  --light-accent4: #1FC598;

  --light-accent5: #53A6E1;

  --light-accent6: #BF65F0;

  --light-accent7: #EE4EB8;

  --light-shade0: #FFFCFF;

  --light-shade1: #E0DCE0;

  --light-shade2: #C1BCC2;

  --light-shade3: #A29DA3;

  --light-shade4: #847E85;

  --light-shade5: #656066;

  --light-shade6: #474247;

  --light-shade7: #282629;
}"
`;

exports[`themer css theme generator should generate valid CSS 2`] = `
"
:root {
  --dark-accent0: rgb(255, 64, 80);
  --dark-accent0-r: 255;
  --dark-accent0-g: 64;
  --dark-accent0-b: 80;

  --dark-accent1: rgb(242, 129, 68);
  --dark-accent1-r: 242;
  --dark-accent1-g: 129;
  --dark-accent1-b: 68;

  --dark-accent2: rgb(255, 210, 74);
  --dark-accent2-r: 255;
  --dark-accent2-g: 210;
  --dark-accent2-b: 74;

  --dark-accent3: rgb(164, 204, 53);
  --dark-accent3-r: 164;
  --dark-accent3-g: 204;
  --dark-accent3-b: 53;

  --dark-accent4: rgb(38, 201, 158);
  --dark-accent4-r: 38;
  --dark-accent4-g: 201;
  --dark-accent4-b: 158;

  --dark-accent5: rgb(102, 191, 255);
  --dark-accent5-r: 102;
  --dark-accent5-g: 191;
  --dark-accent5-b: 255;

  --dark-accent6: rgb(204, 120, 250);
  --dark-accent6-r: 204;
  --dark-accent6-g: 120;
  --dark-accent6-b: 250;

  --dark-accent7: rgb(245, 83, 191);
  --dark-accent7-r: 245;
  --dark-accent7-g: 83;
  --dark-accent7-b: 191;

  --dark-shade0: rgb(40, 38, 41);
  --dark-shade0-r: 40;
  --dark-shade0-g: 38;
  --dark-shade0-b: 41;

  --dark-shade1: rgb(71, 66, 71);
  --dark-shade1-r: 71;
  --dark-shade1-g: 66;
  --dark-shade1-b: 71;

  --dark-shade2: rgb(101, 96, 102);
  --dark-shade2-r: 101;
  --dark-shade2-g: 96;
  --dark-shade2-b: 102;

  --dark-shade3: rgb(132, 126, 133);
  --dark-shade3-r: 132;
  --dark-shade3-g: 126;
  --dark-shade3-b: 133;

  --dark-shade4: rgb(162, 157, 163);
  --dark-shade4-r: 162;
  --dark-shade4-g: 157;
  --dark-shade4-b: 163;

  --dark-shade5: rgb(193, 188, 194);
  --dark-shade5-r: 193;
  --dark-shade5-g: 188;
  --dark-shade5-b: 194;

  --dark-shade6: rgb(224, 220, 224);
  --dark-shade6-r: 224;
  --dark-shade6-g: 220;
  --dark-shade6-b: 224;

  --dark-shade7: rgb(255, 252, 255);
  --dark-shade7-r: 255;
  --dark-shade7-g: 252;
  --dark-shade7-b: 255;

  --light-accent0: rgb(240, 62, 77);
  --light-accent0-r: 240;
  --light-accent0-g: 62;
  --light-accent0-b: 77;

  --light-accent1: rgb(243, 119, 53);
  --light-accent1-r: 243;
  --light-accent1-g: 119;
  --light-accent1-b: 53;

  --light-accent2: rgb(238, 186, 33);
  --light-accent2-r: 238;
  --light-accent2-g: 186;
  --light-accent2-b: 33;

  --light-accent3: rgb(151, 189, 45);
  --light-accent3-r: 151;
  --light-accent3-g: 189;
  --light-accent3-b: 45;

  --light-accent4: rgb(31, 197, 152);
  --light-accent4-r: 31;
  --light-accent4-g: 197;
  --light-accent4-b: 152;

  --light-accent5: rgb(83, 166, 225);
  --light-accent5-r: 83;
  --light-accent5-g: 166;
  --light-accent5-b: 225;

  --light-accent6: rgb(191, 101, 240);
  --light-accent6-r: 191;
  --light-accent6-g: 101;
  --light-accent6-b: 240;

  --light-accent7: rgb(238, 78, 184);
  --light-accent7-r: 238;
  --light-accent7-g: 78;
  --light-accent7-b: 184;

  --light-shade0: rgb(255, 252, 255);
  --light-shade0-r: 255;
  --light-shade0-g: 252;
  --light-shade0-b: 255;

  --light-shade1: rgb(224, 220, 224);
  --light-shade1-r: 224;
  --light-shade1-g: 220;
  --light-shade1-b: 224;

  --light-shade2: rgb(193, 188, 194);
  --light-shade2-r: 193;
  --light-shade2-g: 188;
  --light-shade2-b: 194;

  --light-shade3: rgb(162, 157, 163);
  --light-shade3-r: 162;
  --light-shade3-g: 157;
  --light-shade3-b: 163;

  --light-shade4: rgb(132, 126, 133);
  --light-shade4-r: 132;
  --light-shade4-g: 126;
  --light-shade4-b: 133;

  --light-shade5: rgb(101, 96, 102);
  --light-shade5-r: 101;
  --light-shade5-g: 96;
  --light-shade5-b: 102;

  --light-shade6: rgb(71, 66, 71);
  --light-shade6-r: 71;
  --light-shade6-g: 66;
  --light-shade6-b: 71;

  --light-shade7: rgb(40, 38, 41);
  --light-shade7-r: 40;
  --light-shade7-g: 38;
  --light-shade7-b: 41;
}"
`;

exports[`themer css theme generator should generate valid CSS 3`] = `
"
:root {
  --dark-accent0: hsl(355, 100%, 63%);
  --dark-accent0-h: 355;
  --dark-accent0-s: 100%;
  --dark-accent0-l: 63%;

  --dark-accent1: hsl(21, 87%, 61%);
  --dark-accent1-h: 21;
  --dark-accent1-s: 87%;
  --dark-accent1-l: 61%;

  --dark-accent2: hsl(45, 100%, 65%);
  --dark-accent2-h: 45;
  --dark-accent2-s: 100%;
  --dark-accent2-l: 65%;

  --dark-accent3: hsl(76, 60%, 50%);
  --dark-accent3-h: 76;
  --dark-accent3-s: 60%;
  --dark-accent3-l: 50%;

  --dark-accent4: hsl(164, 68%, 47%);
  --dark-accent4-h: 164;
  --dark-accent4-s: 68%;
  --dark-accent4-l: 47%;

  --dark-accent5: hsl(205, 100%, 70%);
  --dark-accent5-h: 205;
  --dark-accent5-s: 100%;
  --dark-accent5-l: 70%;

  --dark-accent6: hsl(279, 93%, 73%);
  --dark-accent6-h: 279;
  --dark-accent6-s: 93%;
  --dark-accent6-l: 73%;

  --dark-accent7: hsl(320, 89%, 64%);
  --dark-accent7-h: 320;
  --dark-accent7-s: 89%;
  --dark-accent7-l: 64%;

  --dark-shade0: hsl(280, 4%, 15%);
  --dark-shade0-h: 280;
  --dark-shade0-s: 4%;
  --dark-shade0-l: 15%;

  --dark-shade1: hsl(300, 4%, 27%);
  --dark-shade1-h: 300;
  --dark-shade1-s: 4%;
  --dark-shade1-l: 27%;

  --dark-shade2: hsl(290, 3%, 39%);
  --dark-shade2-h: 290;
  --dark-shade2-s: 3%;
  --dark-shade2-l: 39%;

  --dark-shade3: hsl(291, 3%, 51%);
  --dark-shade3-h: 291;
  --dark-shade3-s: 3%;
  --dark-shade3-l: 51%;

  --dark-shade4: hsl(290, 3%, 63%);
  --dark-shade4-h: 290;
  --dark-shade4-s: 3%;
  --dark-shade4-l: 63%;

  --dark-shade5: hsl(290, 5%, 75%);
  --dark-shade5-h: 290;
  --dark-shade5-s: 5%;
  --dark-shade5-l: 75%;

  --dark-shade6: hsl(300, 6%, 87%);
  --dark-shade6-h: 300;
  --dark-shade6-s: 6%;
  --dark-shade6-l: 87%;

  --dark-shade7: hsl(300, 100%, 99%);
  --dark-shade7-h: 300;
  --dark-shade7-s: 100%;
  --dark-shade7-l: 99%;

  --light-accent0: hsl(355, 86%, 59%);
  --light-accent0-h: 355;
  --light-accent0-s: 86%;
  --light-accent0-l: 59%;

  --light-accent1: hsl(21, 89%, 58%);
  --light-accent1-h: 21;
  --light-accent1-s: 89%;
  --light-accent1-l: 58%;

  --light-accent2: hsl(45, 86%, 53%);
  --light-accent2-h: 45;
  --light-accent2-s: 86%;
  --light-accent2-l: 53%;

  --light-accent3: hsl(76, 62%, 46%);
  --light-accent3-h: 76;
  --light-accent3-s: 62%;
  --light-accent3-l: 46%;

  --light-accent4: hsl(164, 73%, 45%);
  --light-accent4-h: 164;
  --light-accent4-s: 73%;
  --light-accent4-l: 45%;

  --light-accent5: hsl(205, 70%, 60%);
  --light-accent5-h: 205;
  --light-accent5-s: 70%;
  --light-accent5-l: 60%;

  --light-accent6: hsl(279, 82%, 67%);
  --light-accent6-h: 279;
  --light-accent6-s: 82%;
  --light-accent6-l: 67%;

  --light-accent7: hsl(320, 82%, 62%);
  --light-accent7-h: 320;
  --light-accent7-s: 82%;
  --light-accent7-l: 62%;

  --light-shade0: hsl(300, 100%, 99%);
  --light-shade0-h: 300;
  --light-shade0-s: 100%;
  --light-shade0-l: 99%;

  --light-shade1: hsl(300, 6%, 87%);
  --light-shade1-h: 300;
  --light-shade1-s: 6%;
  --light-shade1-l: 87%;

  --light-shade2: hsl(290, 5%, 75%);
  --light-shade2-h: 290;
  --light-shade2-s: 5%;
  --light-shade2-l: 75%;

  --light-shade3: hsl(290, 3%, 63%);
  --light-shade3-h: 290;
  --light-shade3-s: 3%;
  --light-shade3-l: 63%;

  --light-shade4: hsl(291, 3%, 51%);
  --light-shade4-h: 291;
  --light-shade4-s: 3%;
  --light-shade4-l: 51%;

  --light-shade5: hsl(290, 3%, 39%);
  --light-shade5-h: 290;
  --light-shade5-s: 3%;
  --light-shade5-l: 39%;

  --light-shade6: hsl(300, 4%, 27%);
  --light-shade6-h: 300;
  --light-shade6-s: 4%;
  --light-shade6-l: 27%;

  --light-shade7: hsl(280, 4%, 15%);
  --light-shade7-h: 280;
  --light-shade7-s: 4%;
  --light-shade7-l: 15%;
}"
`;

exports[`themer css theme generator should provide usage instructions 1`] = `
"
Import the generated theme file into your stylesheet via \`@import()\`, or into your HTML markup via \`<link>\`.

\`hex.css\` provides the theme colors in hex format; \`rgb.css\` and \`hsl.css\` in RGB and HSL formats respectively along with individual channel values for further manipulation if desired.

Generated files:

* \`hex.css\`
* \`rgb.css\`
* \`hsl.css\`
"
`;
